src/utils.sh

```diff
@@ -440,3 +453,41 @@ function handlers_init() {
 }
 
 export -f handlers_init
+
+# Constants
+InstrumentationKey="e83c2a5c-35d1-4a37-a5fe-b4f0f6e8e2a5"
+IngestionEndpoint="https://dc.services.visualstudio.com/v2/track"
+EventName="Azure-IoT-Edge-Installer-Summary"
+DeviceUniqueID="xinzedPC"
+CurrentTime=$(echo `date '+%Y-%m-%dT%H:%M:%S.%N'`)
+SchemaVersion="1.0"
+
+######################################
+# send_appinsight_telemetry
+#
+#    Send Application Insights event as a REST POST request with JSON body containing telemetry
+#
+# ARGUMENTS:
+#    CustomProperties: "status":[EXIT_CODE]
+#    CustomMeasurements: "duration":    
+#
+# OUTPUTS:
+#    Write output to stdout
+# RETURN:
+#
+######################################
+
+function send_appinsight_telemetry ()
+{
+    local customPropertiesObj='"status":'$1''
+
+    # validate that the user has opted in for telemetry collection
+    local optin=$(get_opt_in_selection )
+    if [[ $optin == true ]];
+    then
+        wget --header='Content-Type: application/json' --header='Accept-Charset: UTF-8' --post-data '{"name":"Microsoft.ApplicationInsights.'$InstrumentationKey'.Event","time": "'$CurrentTime'","iKey": "'$InstrumentationKey'","tags":{"ai.cloud.roleInstance": "'$DeviceUniqueID'"},"data":{"baseType": "EventData","baseData": {"ver": "'$SchemaVersion'","name": "'$EventName'","cv": "'$CORRELATION_VECTOR'","properties":{'$customPropertiesObj'},"measurements":{"duration":57}}}}' $IngestionEndpoint
```

Comment:
I would place the generation of CurrentTime here, immediately before the wget call?

Reply:
addressed in new commit